src/api/routes/webhooks/#webhook_id/#token/index.ts

```
@@ -75,16 +77,524 @@ const messageUpload = multer({
+		attachments,
+		timestamp: new Date(),
+	});
+
+	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+	//@ts-ignore dont care2
+	message.edited_timestamp = null;
+
+	webhook.channel.last_message_id = message.id;
+
+	await Promise.all([
+		message.save(),
+		webhook.channel.save(),
+		emitEvent({
+			event: "MESSAGE_CREATE",
+			channel_id: webhook.channel_id,
+			data: message,
+		} as MessageCreateEvent),
+	]);
+
+	// no await as it shouldnt block the message send function and silently catch error
+	postHandleMessage(message).catch((e) =>
+		console.error("[Message] post-message handler failed", e),
+	);
+
+	return res.json(message);
+};
+
+const parseGitHubWebhook = (
+	req: Request,
+	res: Response,
+	next: NextFunction,
+) => {
+	const eventType = req.headers["x-github-event"] as string;
+	if (!eventType) {
+		throw new HTTPError("Missing X-GitHub-Event header", 400);
+	}
+
+	if (eventType === "ping") {
+		return res.status(200).json({ message: "pong" });
+	}
+
+	const discordPayload = transformGitHubToDiscord(eventType, req.body);
+	if (!discordPayload) {
+		// Unsupported event type
+		return res.status(204).send();
+	}
+
+	req.body = discordPayload;
+	console.dir(req.body, { depth: null });
+	// Set default wait=true for GitHub webhooks so they get a response
+	req.query.wait = req.query.wait || "true";
+	next();
+};
+
+function transformGitHubToDiscord(
+	eventType: string,
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	payload: any,
+): WebhookExecuteSchema | null {
+	switch (eventType) {
+		case "star":
+			return {
+				username: "GitHub",
+				// TODO: Provide a static avatar for GitHub
+				embeds: [
+					{
+						title: `⭐ New star on ${payload.repository?.full_name}`,
+						type: EmbedType.rich,
+						description: `${payload.sender?.login} starred the repository`,
+						color: 0xffd700,
+						thumbnail: {
+							url: payload.sender?.avatar_url,
+						},
+						timestamp: new Date().toISOString(),
```

I'd prefer to *not* convert these Dates to strings, however the message won't send unless it's a string. Can't figure out why.

---

This was silently failing when validating against `WebhookExecuteSchema` if using the Date type. That's why strings are needed here. Should be fine.